src/Craftify.Geometry.Domain/LineSegmentIntersections.cs

```diff
@@ -0,0 +1,88 @@
+﻿namespace Craftify.Geometry.Domain;
+
+public static class LineSegmentIntersections
+{
+    public static bool Intersects(this LineSegment main, LineSegment other, out Point3D? intersectionPoint)
+    {
+        intersectionPoint = null;
+
+        Vector3D mainDirectionVector = main.GetDirection();
```

Comment:
I think we should maintain consistency with the entire codebase and use var instead of explicitly specifying the type.